### ISSUE TITLE
clarifying the `await*` example's processing model

### ISSUE THREAD

**Issue by getify** [tc39/ecmascript-asyncawait#8]
In the main README, [a suggestion is made](https://github.com/lukehoban/ecmascript-asyncawait#await-and-parallelism) about a possible future (beyond this spec) `await*` syntax, and this is the example given:

``` js
async function getData() {
  var items = await fetchAsync('http://example.com/users');
  return await* items.map(async(item) => {
    return {
      title: item.title, 
      img: (await fetchAsync(item.userDataUrl)).img
    }
  }
}
```

The example specifically shows the use of `map(..)`, which leads me to assume that this is _the mechanism_ by which all the callback-invocation promises are "transported" back to the hidden `Promise.all` in `await*`. That is, it implies that we have an array of promises returned back from `map()`, and _that's_ how `await*` can consume them.

But what about the other iterators, which don't do this mapping, and don't even necessarily return an array themselves (though they all operate on arrays), like: `forEach(..)`, `some(..)`, `filter(..)`, `reduce(..)`, etc? Is there anyway `await*` could be used with them? 

It would require a deeper, special integration between `await*` and these iterators, where the internal list of those promises from each iteration invocation wouldn't be "discarded" (due to the semantics of each iterator), but could somehow be alternatively tracked internally in the iterator, and then "transported" back to the `await*`. For everything but `map(..)`, this would have to be a different mechanism than just the simple return value array that `map(..)` gives.

To put it another way, instead of the processing model being:
1. run the `map(..)` iteration on the right-hand side, then
2. take the array of promises and pump it into `await*`

...the processing model would/could be a little more like `new fn(..)`, where the `new` operator changes how the `fn(..)` invocation occurs:
1. `await*` could/would change how the right-hand side is processed, such that the special behavior I'm suggesting would be possible.

Is this just reaching too far? I can think of several cases where, for instance, it would be nice to be able to use `await*` on `filter(..)`'d arrays, to replace various libraries that provide async versions of these various iterators for you.
